src/app/services/training/service.py

```diff
@@ -84,14 +69,15 @@ def build_pipeline(
             memory=memory,
         )
 
-    def train(self, file: UploadFile) -> MLModel:
-        X, y = run_pipeline(
-            file=file,
-            filter_config=self.filter_config,
-            mapping_config=self.mapping_config,
-            feature_engineer_config=self.feature_engineer_config,
-            selection_config=self.selection_config,
-        )
+    def train(self, training_file_path: Path) -> MLModel:
+
+        self.file_path = training_file_path
+
+        if not self.file_path.exists():
+            msg = f"Training file not found at {self.file_path}"
+            raise FileNotFoundError(msg)
+
+        self._create_uploadfile_from_path(self.file_path, self.file_path.name)
```

Comment:
this method it is implemented somewhere? or will be implemented later?

Reply:
In addition to this I found this code maybe you can refactor, please evaluate it, and move forward if you think the scenario could occurs:
Check the dataset supplies each expected column, then build the numeric list with order-preserving list comprehensions.

```suggestion
        // ...existing code...
                df = self._load_training_data(file_path)
                expected = set(ohe_cols + ordinal_cols)
                missing = [col for col in expected if col not in df.columns]
                if missing:
                    msg = f"Missing required columns: {', '.join(missing)}"
                    raise ValueError(msg)

                numeric_cols = [
                    col for col in df.columns
                    if col not in ohe_cols and col not in ordinal_cols
                ]
        // ...existing code...
```
This consideration, could not be the best approach, allows to handle missing columns and ensure numeric/ordinal splits stay ordered.